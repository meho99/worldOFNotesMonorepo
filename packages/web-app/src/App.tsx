import React from 'react'

export const App = () => {
  return (
    <div>Test</div>
  )
}
App.displayName = 'AppComponent'
