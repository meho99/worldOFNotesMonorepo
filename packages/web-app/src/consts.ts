export enum keyNames {
  Session = 'session',
  Notes = 'notes',
  Router = 'router'
} 
