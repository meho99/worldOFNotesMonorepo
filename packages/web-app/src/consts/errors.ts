export enum FieldsErrors {
  PasswordToShort = 'Password must have at least 8 characters',
  PasswordToLong = 'Password must have less than 8 characters',
  NameToShort = 'Name must have at least 3 characters',
  NameToLong = 'Name must have less than 25 characters',
  Required = 'Required field',
  InvalidEmailFormat = 'Invalid email address'
}

export enum FetchingErrors {
  LoginError = 'Unable to login. Please check your email and password and try again.',
  SingUpError = 'Unable to sign up. Please check your email and password and try again.',
  Unauthorized = 'Unauthorized'
}
