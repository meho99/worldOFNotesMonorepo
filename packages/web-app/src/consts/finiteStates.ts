export enum FiniteStates {
  Idle = 'idle',
  Loading = 'loading',
  Success = 'success',
  Failure = 'failure',
}
