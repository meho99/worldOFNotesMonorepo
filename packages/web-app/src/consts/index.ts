export * from './urls'
export * from './themeTypes'
export * from './finiteStates'
export * from './reducerNames'
export * from './errors'
export * from './notificationTypes'
