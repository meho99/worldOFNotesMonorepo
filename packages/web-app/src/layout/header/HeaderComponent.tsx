import React from 'react'
import { AppBar, Toolbar, Typography } from '@material-ui/core'

import { useStyles } from './Header.styles'
import { HeaderMenu } from './HeaderMenu'

export const HeaderComponent = () => {
  const classes = useStyles()

  return (
    <AppBar position='fixed' color='secondary'>
      <Toolbar>
        <Typography variant='h6' className={classes.title}>
          World of Notes
        </Typography>

        <HeaderMenu />
      </Toolbar>
    </AppBar>
  )
}
