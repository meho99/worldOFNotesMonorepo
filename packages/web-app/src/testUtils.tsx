import React from 'react'
import { DeepPartial } from 'redux'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import { ReducerState } from './redux/reducers'

const mockStore = configureStore()

export const CreateProvider = (testState: DeepPartial<ReducerState>): React.FC => ({ children }) =>
  <Provider store={mockStore(testState)}> {children} </Provider>
