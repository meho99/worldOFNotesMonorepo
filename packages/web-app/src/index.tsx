import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import { createGenerateClassName, StylesProvider } from '@material-ui/core/styles'

import { store } from './redux'
import { ThemeProvider } from './theme'

import { App } from './App'

const generateClassName = createGenerateClassName({
  productionPrefix: 'won',
  disableGlobal: true
})

ReactDOM.render(
  <StylesProvider generateClassName={generateClassName} injectFirst>
    <ThemeProvider variant='dark'>
      <Provider store={store}>
        <App />
      </Provider>
    </ThemeProvider>
  </StylesProvider>,
  document.getElementById('root')
)
