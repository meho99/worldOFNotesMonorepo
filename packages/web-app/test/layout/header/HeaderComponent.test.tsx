import React from 'react'
import configureStore from 'redux-mock-store'
import { render } from '@testing-library/react'

import { HeaderComponent } from '../../../src/layout/header/HeaderComponent'
import { FiniteStates } from '../../../src/consts/finiteStates'
import { CreateProvider } from '../../../src/testUtils'

describe('HeaderComponent test', () => {configureStore
  it('Snapshot test', () => {
    expect(render(<HeaderComponent />, {
      wrapper: CreateProvider({
        session: {
          authenticatingStatus: FiniteStates.Success
        }
      })
    }).container).toMatchSnapshot()
  })
})
