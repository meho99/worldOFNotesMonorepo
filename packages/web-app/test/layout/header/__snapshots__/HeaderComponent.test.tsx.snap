// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent test Snapshot test 1`] = `
<div>
   
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
      >
        World of Notes
      </h6>
      <div>
        <button
          aria-controls="simple-menu"
          aria-haspopup="true"
          aria-label="menu"
          class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </header>
   
</div>
`;
